### ISSUE TITLE
[FEATURE] Generalize DataSourcePicker/datasourceid usage

### ISSUE THREAD

**Issue by derek-ho** [opensearch-project/security-dashboards-plugin#1816]
**Is your feature request related to a problem?**
- After this pr is merged: https://github.com/opensearch-project/security-dashboards-plugin/pull/1810, there is some cleanup that is necessary:
- Hook up selected datasource to a global state/context/store that can be read and modified on each tab - investigate options here
- Clean up datasource picker registration (currently it would be globally registered), need to figure out how to register it only in our plugin, be able to change it from writeable to read only on different screens
- Add testing framework/e2e testing framework to verify that correct API calls are being made - maybe blocked on https://github.com/opensearch-project/OpenSearch-Dashboards/issues/4524 - possible workaround is to run with security on ```http://``` instead of ```https://```

**Comment by derek-ho** [comment]
Comprehensive Cypress tests will be blocked on: https://github.com/opensearch-project/security-dashboards-plugin/issues/1815 - we need some indication of success or failure that we can rely on when running cypress tests, since it will be difficult/impossible to verify that the backend has received/executed the call correctly in github runners cypress testing.

**Comment by cwperks** [comment]
[Triage] @derek-ho Can you please add acceptance criteria on this issue? Removing the untriaged label.

**Comment by derek-ho** [comment]
Acceptance criteria:
- Decide on the approach out of two current options to maintain dataSourceId context within our plugin: local/session storage and useState/useContext that gets passed down into the tabs.
- Implement that decision in the context of the requests for two tabs

**Comment by derek-ho** [comment]
Merged into feature branch
